WebDriverTests/src/test/java/Task40/FrameTest.java

```diff
@@ -0,0 +1,80 @@
+package Task40;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.interactions.Actions;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class FrameTest {
+    private static final String DESTINATION_URL = "https://the-internet.herokuapp.com/iframe";
+    private static final String FRAME_LOCATOR = "mce_0_ifr";
+    private static final String TEXT = "Hello world!";
+    private static final String BOLD_TEXT= "world!";
+    private static final By TEXT_EDITOR_LOCATOR = By.id("tinymce");
+    private static final By BUTTON_BOLD_LOCATOR = By.cssSelector("#mceu_3 > button");
+    private static final By TEXT_ELEMENT_LOCATOR = By.cssSelector("#tinymce > p");
+    private static final By BOLD_TEXT_LOCATOR = By.cssSelector("#tinymce > p > strong");
+    WebDriver driver;
+    WebElement textEditor;
+    Actions action;
+
+    @BeforeMethod
+    public void startup() {
+        System.setProperty("webdriver.chrome.driver", "src/test/resources/chromedriver.exe");
+        driver = new ChromeDriver();
+        driver.manage().window().maximize();
+        driver.get(DESTINATION_URL);
+
+        driver.switchTo().frame(FRAME_LOCATOR);
+        textEditor = driver.findElement(TEXT_EDITOR_LOCATOR);
+        textEditor.clear();
+    }
+
+    @AfterMethod
+    public void closeBrowser() {
+        driver.close();
+    }
+
+    @Test(groups = { "regression", "frame" })
+    public void verifyAddedText() {
+        textEditor.sendKeys(TEXT);
+        selectText();
+        driver.switchTo().defaultContent();
+        driver.findElement(BUTTON_BOLD_LOCATOR).click();
+        driver.switchTo().frame(FRAME_LOCATOR);
+        String textResult = driver.findElement(TEXT_ELEMENT_LOCATOR).getText();
+        String bolTextResult = driver.findElement(BOLD_TEXT_LOCATOR).getText();
+
+        Assert.assertEquals(TEXT, textResult, "Expected text is '" + TEXT + "' but actual text is '" + textResult + "'.");
```

Comment:
When you need to perform multiple assertions at the end, use SoftAssert. It has method assertAll(), which we can execute at the end and see, what is incorrect. But do not use it to assert something at the end, then in middle and so on.